src/config/mod.rs

```diff
@@ -3571,44 +3564,42 @@ impl TikvConfig {
         config::canonicalize_sub_path(data_dir, DEFAULT_ROCKSDB_SUB_DIR)
     }
 
-    pub fn validate(&mut self) -> Result<(), Box<dyn Error>> {
-        // Setting up data paths.
-        if self.cfg_path.is_empty() {
-            self.cfg_path = Path::new(&self.storage.data_dir)
-                .join(LAST_CONFIG_FILE)
-                .to_str()
-                .unwrap()
-                .to_owned();
-        }
-
-        match (
-            self.raft_store.raftdb_path.is_empty(),
-            self.raft_engine.config.dir.is_empty(),
-        ) {
-            (false, true) => {
-                // If raftdb_path is specified, raft_engine_path will inherit it, this will be
-                // useful when updating from older version.
-                self.raft_engine.config.dir =
-                    self.infer_raft_engine_path(Some(self.raft_store.raftdb_path.as_str()))?;
-                self.raft_store.raftdb_path = self.infer_raft_db_path(None)?;
-            }
-            _ => {
-                self.raft_store.raftdb_path = self.infer_raft_db_path(None)?;
-                self.raft_engine.config.dir = self.infer_raft_engine_path(None)?;
+    fn titan_data_exists(&self) -> Result<bool, Box<dyn Error>> {
```

Comment:
Somehow the difference in UI is unnecessarily too big. Can it be made to only show real difference?